### PR TITLE
[sw] Clean up example device programs

### PR DESCRIPTION
**Pull request by mcy**
This makes the two "hello" examples a little more readable, and factors out some common demos in them, as well.

It also removes a couple of nasty warts in hello_usbdev, including a very very UB alignment hack (which is now handled, as it should be, by the linker).

### CODE REVIEW

**Comment by mcy** [comment]
Addressed the demos comment.

File: sw/device/exts/common/flash_link.ld [hunk 1]
```diff
@@ -96,6 +96,12 @@ SECTIONS {
     *(.data.*)
     *(.sdata)
     _data_end = .;
+    /**
+     * The VMem format is composed of 32-bit words. Since |.data| may contain
+     * symbols that do not end on an alignment boundary, we pad it out to avoid
+     * any issues.
+     */
+    . = ALIGN(4);
```
Comment (gkelly) [inline sw/device/exts/common/flash_link.ld]: This is a separate bugfix, put it into a separate commit please.

Comment (mcy) [reply sw/device/exts/common/flash_link.ld]: Done.
